cmd/gateway-s3.go

```diff
@@ -450,7 +454,11 @@ func toMinioClientMetadata(metadata map[string]string) map[string]string {
 func (l *s3Objects) NewMultipartUpload(bucket string, object string, metadata map[string]string) (uploadID string, err error) {
 	// Create PutObject options
 	opts := minio.PutObjectOptions{UserMetadata: metadata}
-	return l.Client.NewMultipartUpload(bucket, object, opts)
+	uploadID, err = l.Client.NewMultipartUpload(bucket, object, opts)
+	if err != nil {
+		return uploadID, s3ToObjectError(traceError(err), bucket, object)
+	}
+	return uploadID, nil
```

Comment:
The if-block can be avoided. A single return statement would be enough as `s3ToObjectError` would return nil of the error is nil.